src/lib/components/workflow/client-actions/update-confirmation-modal.svelte

```diff
@@ -0,0 +1,182 @@
+<script lang="ts">
+  import { writable, type Writable } from 'svelte/store';
+
+  import { v4 as uuid } from 'uuid';
+
+  import { page } from '$app/stores';
+
+  import PayloadDecoder from '$lib/components/event/payload-decoder.svelte';
+  import { type PayloadInputEncoding } from '$lib/components/payload-input-with-encoding.svelte';
+  import PayloadInput from '$lib/components/payload-input.svelte';
+  import Alert from '$lib/holocene/alert.svelte';
+  import Button from '$lib/holocene/button.svelte';
+  import CodeBlock from '$lib/holocene/code-block.svelte';
+  import Input from '$lib/holocene/input/input.svelte';
+  import Modal from '$lib/holocene/modal.svelte';
+  import Option from '$lib/holocene/select/option.svelte';
+  import Select from '$lib/holocene/select/select.svelte';
+  import { translate } from '$lib/i18n/translate';
+  import { updateWorkflow } from '$lib/services/workflow-service';
+  import { toaster } from '$lib/stores/toaster';
+  import { workflowRun } from '$lib/stores/workflow-run';
+  import type { WorkflowExecution } from '$lib/types/workflows';
+  import { isNetworkError } from '$lib/utilities/is-network-error';
+
+  $: ({ run: runId } = $page.params);
+  $: ({ metadata } = $workflowRun);
+  $: updateDefinitions = metadata?.definition?.updateDefinitions;
+
+  export let open: boolean;
+  export let workflow: WorkflowExecution;
+  export let namespace: string;
+
+  const defaultEncoding: PayloadInputEncoding = 'json/plain';
+
+  let error = '';
+  let loading = false;
+  let failure;
+  let success;
+
+  let name = '';
+  let updateId = uuid();
+  let input = '';
+  let customUpdate = false;
+  let encoding: Writable<PayloadInputEncoding> = writable(defaultEncoding);
+
+  const hideModal = () => {
+    open = false;
+    name = '';
+    input = '';
+    customUpdate = false;
+    $encoding = defaultEncoding;
+    failure = undefined;
+    success = undefined;
+  };
+
+  const update = async () => {
+    failure = undefined;
+    success = undefined;
+    error = '';
+    loading = true;
+
+    try {
+      const result = await updateWorkflow({
+        namespace,
+        workflow: { workflowId: workflow.id, runId },
+        input,
+        updateId,
+        encoding: $encoding,
+        name,
+      });
+
+      failure = result?.outcome?.failure;
+      success = result?.outcome?.success;
+      updateId = uuid();
+
+      if (success) {
+        toaster.push({
+          message: translate('workflows.update-success'),
+          variant: 'success',
+        });
+        hideModal();
+      }
```

Comment:
Looks like there's an edge case here where it doesn't error but isn't successful and the modal remains open with no error after update ⬇️ 

| During Update | After Update |
|-|-|
|![Screenshot 2025-01-21 at 3 05 46 PM](https://github.com/user-attachments/assets/95ccd329-976e-4808-b998-ce98c37ac288)|![Screenshot 2025-01-21 at 3 06 05 PM](https://github.com/user-attachments/assets/95f8e3d7-2622-43fe-b38e-d505eacfb198)|

This is what the result looks like, it appears not to have an `outcome` ⬇️ 

![Screenshot 2025-01-21 at 3 16 25 PM](https://github.com/user-attachments/assets/3f376ccd-703b-406e-9548-fe4279ebaaab)

Reply:
👍 added check for success if also not a failure